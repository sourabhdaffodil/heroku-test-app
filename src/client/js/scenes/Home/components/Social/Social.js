import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { media } from '../../../../theme/style-utils';
import Icon from '../../../../components/Icon/Icon';
import Link from '../../../../components/Link/Link';

const Wrapper = styled.div`
  > a {
    margin: 0 10px;
    
    ${media.phone`
      display: inline-block;
      width: 45px;
      text-align: center;
    `}
  }
  > a:last-child {
    margin-right: 0;
  }
`;

class Social extends Component {
  renderListItem() {
    return this.props.social.map(social => (
      <Link key={social.network} url={social.url}>
        <Icon icon={`${social.network} fa-2x`} />
      </Link>
    ));
  }

  render() {
    return (
      <Wrapper>
        {this.renderListItem()}
      </Wrapper>
    );
  }
}

Social.propTypes = {
  social: PropTypes.array.isRequired,
};

function mapStateToProps(state) {
  return {
    social: state.about.social,
  };
}

export default connect(mapStateToProps)(Social);

export {
  mapStateToProps,
  Social,
};
