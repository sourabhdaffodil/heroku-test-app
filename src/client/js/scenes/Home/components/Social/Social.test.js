import React from 'react';
import { shallow } from 'enzyme';
import { Social, mapStateToProps } from './Social';

describe('Social icons component', () => {
  const social = [
    {
      network: 'facebook',
      url: 'https://www.facebook.com/orenciorodolfo',
    },
    {
      network: 'linkedin',
      url: 'https://www.linkedin.com/in/rodolfo-goncalves',
    },
  ];

  it('Must render an empty list properly', () => {
    const wrapper = shallow(
      <Social social={[]} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Must render a list of social links with icons', () => {
    const wrapper = shallow(
      <Social social={social} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('Social mapStateToProps function', () => {
    it('Must return an object with an empty "social" property', () => {
      const defaultState = {
        about: {
          social: [],
        },
      };
      const result = mapStateToProps(defaultState);
      const expected = {
        social: [],
      };
      expect(result).toEqual(expected);
    });

    it('Must map the state "social" to the component\'s properties', () => {
      const result = mapStateToProps({ about: { social } });
      const expected = {
        social,
      };
      expect(result).toEqual(expected);
    });
  });
});
