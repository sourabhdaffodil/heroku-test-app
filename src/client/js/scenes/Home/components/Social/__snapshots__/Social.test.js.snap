// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social icons component Must render a list of social links with icons 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Social
    social={
      Array [
        Object {
          "network": "facebook",
          "url": "https://www.facebook.com/orenciorodolfo",
        },
        Object {
          "network": "linkedin",
          "url": "https://www.linkedin.com/in/rodolfo-goncalves",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Link
          className=""
          url="https://www.facebook.com/orenciorodolfo"
        >
          <Icon
            icon="facebook fa-2x"
          />
        </Link>,
        <Link
          className=""
          url="https://www.linkedin.com/in/rodolfo-goncalves"
        >
          <Icon
            icon="linkedin fa-2x"
          />
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "facebook",
        "nodeType": "function",
        "props": Object {
          "children": <Icon
            icon="facebook fa-2x"
          />,
          "className": "",
          "url": "https://www.facebook.com/orenciorodolfo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "icon": "facebook fa-2x",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "linkedin",
        "nodeType": "function",
        "props": Object {
          "children": <Icon
            icon="linkedin fa-2x"
          />,
          "className": "",
          "url": "https://www.linkedin.com/in/rodolfo-goncalves",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "icon": "linkedin fa-2x",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Link
            className=""
            url="https://www.facebook.com/orenciorodolfo"
          >
            <Icon
              icon="facebook fa-2x"
            />
          </Link>,
          <Link
            className=""
            url="https://www.linkedin.com/in/rodolfo-goncalves"
          >
            <Icon
              icon="linkedin fa-2x"
            />
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "facebook",
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              icon="facebook fa-2x"
            />,
            "className": "",
            "url": "https://www.facebook.com/orenciorodolfo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "icon": "facebook fa-2x",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "linkedin",
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              icon="linkedin fa-2x"
            />,
            "className": "",
            "url": "https://www.linkedin.com/in/rodolfo-goncalves",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "icon": "linkedin fa-2x",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Social icons component Must render an empty list properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Social
    social={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
